### DISCUSSION TITLE
How to use PoisoningAttackBackdoor

### DISCUSSION THREAD

**Discussion by schroe-ja** [Trusted-AI/adversarial-robustness-toolbox#1610] [Q&A] [answered]
I want to use the backdoor attack against my own dataset. The notebook only shows a variant with load_mnist(). This didn't really help because everything depends on ART functions.

**Comment by beat-buesser** [comment, marked as answer]
Hi @EvilWatermelon Could you please describe the issue in more detail?

**Reply by schroe-ja** [reply]
For my masterthesis I implement a framework which measure risks of ML models. One of my goals is to attack a ML model and then measure the differences. I want to use ART because there are already implemented backdoor attacks but I have issues to use the attacks of ART.

I tried to implement the attacks with the notebooks and examples. But the examples load training data from special functions `load_mnist()` etc. 

Can I use the attack functions with my own training data or is only possible to load training data from the ART functions?

**Reply by beat-buesser** [reply]
That sounds interesting! You can use any image dateset, you don't have to use ART's utility functions to load data. The attacks usually expect images `x` to be of shape NHWC or NCHW, depending on what your estimator/model is expecting as input shape, as type `np.ndarray`. Let us know if you have more questions!
